generate_engagement_analysis_files.py

```diff
@@ -103,3 +103,60 @@
 
         for row in opted_in_uids_per_show.items():
             writer.writerow(row)
+
+    log.info(f'Computing repeat and new participation per show ...')
+    # Computes the number of new and repeat consented individuals who participated in each radio show.
+    # Repeat participants are consented individuals who participated in previous shows prior to the target show.
+    # New participants are consented individuals who participated in target show but din't participate in previous shows.
+    repeat_new_participation_map = OrderedDict() # of rqa_raw_field to participation metrics.
+    for rqa_raw_field in rqa_raw_fields:
+        target_radio_show = rqa_raw_field # radio show in which we are calculating participation metrics for.
+
+        target_radio_show_uids = [] # contains uids of individuals who participated in target radio show.
+        with open(f'{demog_map_json_input_dir}/{target_radio_show}_demog_map.json') as f:
+            target_radio_show_data = json.load(f)
+            for uid in [*target_radio_show_data]:
+                target_radio_show_uids.append(uid)
+
+        previous_radio_shows = [] # rqa_raw_fields of shows that aired before the target radio show.
+        for rqa_raw_field in rqa_raw_fields:
+            if rqa_raw_field == target_radio_show:
+                break
+            previous_radio_shows.append(rqa_raw_field)
+
+        previous_radio_shows_participants = set() # uids of individuals who participated in previous radio shows.
+        for rqa_raw_field in previous_radio_shows:
+            with open(f'{demog_map_json_input_dir}/{rqa_raw_field}_demog_map.json') as f:
+                previous_radio_shows_data = json.load(f)
+                for uid in [*previous_radio_shows_data]:
+                    previous_radio_shows_participants.add(uid)
+
+        repeat_uids = set() # uids of individuals who participated in target and previous shows.
+        new_uids = set() # uids of individuals who participated in target show but din't participate in previous shows.
+        for uid in target_radio_show_uids:
+            if uid in previous_radio_shows_participants:
+                repeat_uids.add(uid)
+            else:
+                new_uids.add(uid)
+
+        repeat_new_participation_map[rqa_raw_field] ={
+            "Radio Show":rqa_raw_field, # Todo switch to dataset name
+            "No. of participants with opt-ins": len(target_radio_show_uids),
+            "No. of new participants with opt-ins": len(new_uids),
+            "% of new participants with opt-ins": round(
```

Comment:
This reads as if it's the % of new participants who opted in (i.e. consenting / total * 100).

Suggest "% of opt-in participants that are new" or similar

Reply:
updated thanks.